xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/layouts/LivedataLayout.vue

```diff
@@ -34,6 +34,9 @@
 -->
 <template>
   <div class="livedata-layout">
+    <div class="description" :id="descriptionId" v-if="hasDescription">
```

Comment:
Note that `div` has no spacing etc. by default while `p` has which is why I suggested using a `p` - in the screenshot you show, for example, I miss spacing above the caption. I don't know what styling you get by `&:extend(caption);`, though.

Also, I'm wondering if using such generic classes like `description` is a good idea as this way, styles could easily clash with similar elements on other parts of the UI. But I don't know if we have any guidelines on that.

Reply:
Hum indeed, while having short names is fine in combination with vue scope style for local definitions, this is not ok for external css that might have unexpected impacts on the descriptions.

Reply:
> Also, I'm wondering if using such generic classes like description is a good idea as this way, styles could easily clash with similar elements on other parts of the UI. But I don't know if we have any guidelines on that.

From what I can see, this is barely used yet (see https://github.com/search?q=org%3Axwiki++.description+language%3ACSS+OR+language%3ALESS+&type=code) so there shouldn't be any collision. Fields use the `hint` vocabulary instead with classes like `.xHint`.

However I agree that adding more specificity to [the style](https://github.com/xwiki/xwiki-platform/pull/2778/files#diff-85e5e59addf6ae705f1cafde1728bb052d6f4ca0e6e022e10ab3998e162d7c2fR170-R174) will avoid unexpected issues if we ever use this class again later on :+1:

Reply:
What about `livedata-layout-description` or `xwiki-livedata-layout-description`?

Reply:
Using `livedata-layout-description` for now as it is consistent with the `livedata-layout` class of the parent element.